toothpick-runtime/src/main/java/toothpick/configuration/Configuration.java

```diff
@@ -0,0 +1,105 @@
+package toothpick.configuration;
+
+import toothpick.Factory;
+import toothpick.MemberInjector;
+import toothpick.config.Binding;
+
+/**
+ * Strategy pattern that allows to change various behaviors of Toothpick.
+ * The default configuration is {@link #production()} and {@link #reflection()}.
+ * A custom configuration can be created and used by toothpick,
+ * it is even possible to use a composition of the built-in configurations.
+ */
+public class Configuration {
```

Comment:
regarding the final, the user should be able to extend Configuration and provide a custom one.